File: desktop-app/src/main/java/entities/Ranking.java
package entities;

public class Ranking {
    private int id;
    private int fighterId;
    private int disciplineId;
    private int rankPosition;
    private double points;
    private String season;

    public Ranking() {
    }

    public Ranking(int fighterId, int disciplineId, int rankPosition, double points, String season) {
        this.fighterId = fighterId;
        this.disciplineId = disciplineId;
        this.rankPosition = rankPosition;
        this.points = points;
        this.season = season;
    }

    public Ranking(int id, int fighterId, int disciplineId, int rankPosition, double points, String season) {
        this.id = id;
        this.fighterId = fighterId;
        this.disciplineId = disciplineId;
        this.rankPosition = rankPosition;
        this.points = points;
        this.season = season;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getFighterId() {
        return fighterId;
    }

    public void setFighterId(int fighterId) {
        this.fighterId = fighterId;
    }

    public int getDisciplineId() {
        return disciplineId;
    }

    public void setDisciplineId(int disciplineId) {
        this.disciplineId = disciplineId;
    }

    public int getRankPosition() {
        return rankPosition;
    }

    public void setRankPosition(int rankPosition) {
        this.rankPosition = rankPosition;
    }

    public double getPoints() {
        return points;
    }

    public void setPoints(double points) {
        this.points = points;
    }

    public String getSeason() {
        return season;
    }

    public void setSeason(String season) {
        this.season = season;
    }

    @Override
    public String toString() {
        return "Ranking{id=" + id + ", fighterId=" + fighterId + ", rankPosition=" + rankPosition + ", points=" + points
                + ", season='" + season + "'}";
    }
}

File: web-app/smartfight-web/src/Entity/Ranking.php
<?php

namespace App\Entity;

use App\Repository\RankingRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: RankingRepository::class)]
#[ORM\Table(name: 'ranking')]
class Ranking
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'id', type: 'integer')]
    private ?int $id = null;

    #[ORM\Column(name: 'fighter_id', type: 'integer', nullable: true)]
    private ?int $fighterId = null;

    #[ORM\Column(name: 'discipline_id', type: 'integer', nullable: true)]
    private ?int $disciplineId = null;

    #[ORM\Column(name: 'rank_position', type: 'integer', nullable: true)]
    private ?int $rankPosition = null;

    #[ORM\Column(name: 'points', type: 'integer', nullable: true)]
    private ?int $points = null;

    #[ORM\Column(name: 'season', type: 'string', length: 20, nullable: true)]
    private ?string $season = null;

    public function getId(): ?int { return $this->id; }
    public function getFighterId(): ?int { return $this->fighterId; }
    public function setFighterId(?int $fighterId): static { $this->fighterId = $fighterId; return $this; }
    public function getDisciplineId(): ?int { return $this->disciplineId; }
    public function setDisciplineId(?int $disciplineId): static { $this->disciplineId = $disciplineId; return $this; }
    public function getRankPosition(): ?int { return $this->rankPosition; }
    public function setRankPosition(?int $rankPosition): static { $this->rankPosition = $rankPosition; return $this; }
    public function getPoints(): ?int { return $this->points; }
    public function setPoints(?int $points): static { $this->points = $points; return $this; }
    public function getSeason(): ?string { return $this->season; }
    public function setSeason(?string $season): static { $this->season = $season; return $this; }

    public function __toString(): string
    {
        return 'Ranking #' . $this->id;
    }
}
